src/ColorPickerBubbleMenu/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { ColorPickerMode } from "../utils/types";
+
+export interface ColorPickerBubbleMenuProps
+  extends Partial<
+    Except<ControlledBubbleMenuProps, "open" | "editor" | "children">
+  > {
+  /**
+   * Provide default list of colors (must be valid CSS color strings) which
+   * are used to form buttons for color swatches.
+   */
+  swatchColors?: SwatchColorOption[];
+  /** Override the props for the color picker. */
+  ColorPickerProps?: Partial<ColorPickerProps>;
+  /**
+   * Used to indicate the default color of the text in the Tiptap editor, if no
+   * color has been set with the color extension (or if color has been *unset*
+   * with the extension). Typically should be set to the same value as the MUI
+   * `theme.palette.text.primary`.
+   */
+  defaultTextColor?: string;
+  mode?: ColorPickerMode;
+}
+
+// Tiptap will return any textStyle attributes when calling
+// `getAttributes("textStyle")`, but here we care about `color`, so add more
+// explicit typing for that. Based on
+// https://github.com/ueberdosis/tiptap/blob/6cbc2d423391c950558721510c1b4c8614feb534/packages/extension-color/src/color.ts#L37-L51
+interface TextStyleAttrs extends ReturnType<Editor["getAttributes"]> {
+  color?: string | null;
+}
+
+/**
+ * A component that renders a bubble menu for color picker.
+ *
+ * Pairs well with the `<MenuButtonTextColor />`, `<MenuButtonHighlightColor />` component.
+ *
+ * If you're using `RichTextEditor`, include this component via
+ * `RichTextEditor`’s `children` render-prop. Otherwise, include the
+ * `ColorPickerBubbleMenu` as a child of the component where you call `useEditor` and
+ * render your `RichTextField` or `RichTextContent`. (The bubble menu itself
+ * will be positioned appropriately no matter where you put it in your React
+ * tree, as long as it is re-rendered whenever the Tiptap `editor` forces an
+ * update, which will happen if it's a child of the component using
+ * `useEditor`).
+ */
+export default function ColorPickerBubbleMenu({
+  swatchColors,
+  defaultTextColor = "",
+  ColorPickerProps,
+  ...controlledBubbleMenuProps
+}: ColorPickerBubbleMenuProps) {
+  const editor = useRichTextEditorContext();
+
+  if (!editor?.isEditable) {
+    return null;
+  }
+
+  if (!("colorPickerBubbleMenuHandler" in editor.storage)) {
+    throw new Error(
+      "You must add the ColorPickerBubbleMenuHandler extension to the useEditor `extensions` array in order to use this component!"
+    );
+  }
+  const handlerStorage = editor.storage
+    .colorPickerBubbleMenuHandler as ColorPickerBubbleMenuHandlerStorage;
+
+  // Update the menu step if the bubble menu state has changed
+  const menuState = handlerStorage.state;
+
+  // Determine if all of the selected content shares the same set color.
+  const allCurrentTextStyleAttrs: TextStyleAttrs[] =
+    getAttributesForEachSelected(editor.state, "textStyle");
+  const isTextStyleAppliedToEntireSelection = !!editor.isActive("textStyle");
```

Comment:
Unfortunately all of this code is specific to "text color" state (i.e. `@tiptap/extension-color`) and does not handle highlight color (i.e. `@tiptap/extension-highlight`). Notice there's different logic for handling highlight color state in `MenuButtonHighlightColor`:

https://github.com/sjdemartini/mui-tiptap/blob/d3528c742e905109087312addabb5ea9f1911f2c/src/controls/MenuButtonHighlightColor.tsx#L41-L49

As such, even with the `mode` prop, this component actually only seems to properly support text color, and the highlight color menu button does not work with `useWithBubbleMenu={true}`. You can still _set_ the highlight color thanks to the `mode` prop, but the picker state isn't initialized correctly. It will always be initialized with the currently selected _text color_, even if opening the highlight color picker.